File: Angular/src/app/movie.ts
import { Actor } from "./actor";
import { Review } from "./review";

export class Movie {
    id?: number;
    title: string = ''; // Ensure that title is not nullable
    poster: File | undefined; // Match the type with the backend (File or undefined)
    posterPath: string | undefined; // Match the type with the backend (string or undefined)
    releaseDate: Date | undefined; // Match the type with the backend (Date or undefined)
    description: string | undefined; // Match the type with the backend (string or undefined)
    actors: Actor[] | undefined; // Match the type with the backend (Actor[] or undefined)
    reviews: Review[] | undefined; // Match the type with the backend (Review[] or undefined)
  }
  

File: Api/ProjectDemo/ProjectDemo/Core/Domain/Entities/Movie.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Caching.Memory;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProjectDemo.Core.Domain.Entities
{
    public class Movie
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; } 
        public DateTime? ReleaseDate { get; set;}
        [FromForm]
        [NotMapped]
        public IFormFile?Poster { get; set; }
        public string? PosterPath { get;  set; }
        public ICollection<Actor>? Actors { get; set; }
        public ICollection<Review>? Reviews { get; set; }
        [NotMapped]
        public string? ActorsStr { get; set; }
        [NotMapped]
        public string? ReviewsStr { get; set; }
        
    }
}
